javascript/lib.c

```diff
@@ -52,10 +53,105 @@ napi_value FindAPI(napi_env env, napi_callback_info info) {
     return js_result;
 }
 
+size_t count_char(strzl_haystack_t strzl_haystack, char needle) {
+    size_t result = strzl_naive_count_char(strzl_haystack, needle);
+
+    return result;
+}
+
+napi_value CountSubstrAPI(napi_env env, napi_callback_info info) {
+    size_t argc = 3;
+    napi_value args[3];
+    napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+
+    // Extract the C string from the JavaScript string for haystack and needle
+    size_t haystack_l;
+    size_t needle_l;
+
+    // For haystack
+    napi_get_value_string_utf8(env, args[0], NULL, 0, &haystack_l);
+    char *haystack = malloc(haystack_l + 1);
+    napi_get_value_string_utf8(env, args[0], haystack, haystack_l + 1, &needle_l);
+    struct strzl_haystack_t strzl_haystack = {haystack, needle_l};
+
+    // For needle
+    napi_get_value_string_utf8(env, args[1], NULL, 0, &haystack_l);
+    char *needle = malloc(haystack_l + 1);
+    napi_get_value_string_utf8(env, args[1], needle, haystack_l + 1, &needle_l);
+    struct strzl_needle_t strzl_needle = {needle, needle_l, 0};
+
+    bool overlap = false;
+    napi_get_value_bool(env, args[2], &overlap);
```

Comment:
What happens if the user passes just 2 arguments?

Reply:
It works fine, overlap will be false

Reply:
You're probably mistaken. I believe it is an undefined behavior, and you may have been lucky to get the right result. The `argc` will be updated after the `napi_get_cb_info`, and must be checked if 2 or 3 args are passed. The last member of `args` may remain uninitialized, storing the previous state of that memory region.